pkg/cloudprovider/gce/gce.go

```diff
@@ -154,6 +178,7 @@ func newGCECloud(config io.Reader) (*GCECloud, error) {
 		projectID:        projectID,
 		zone:             zone,
 		instanceID:       instanceID,
+		networkName:      networkName,
```

Comment:
It's used for the Routes creation. And you're right: I did actively assume that - does the cluster even work when they are not? How would the master instance contact the node instances?
Anyway, if we do not want that assumption, we probably need to pass in the node network name to the master and into the gce client.